### DISCUSSION TITLE
Kerberos password authentication

### DISCUSSION THREAD

**Discussion by El-Virus** [keycloak/keycloak#19734] [Q&A]
Hi,
I'm attempting to set up Kerberos as a User Federation provider.
I belive I have all the configurations fields set correctly (realm, principal and allow password authentication is enabled.)
The configured keytab file does have the principal details on it, and I can successfully `kinit` as my user in the same server where keycloak is installed.
When attempting to login via the web console, keycloak gives the following output:
```
Debug is  true storeKey true useTicketCache false useKeyTab false doNotPrompt false ticketCache is null isInitiator true KeyTab is null refreshKrb5Config is false principal is null tryFirstPass is false useFirstPass is false storePass is false clearPass is false
		[Krb5LoginModule] user entered username: elvirus@<DOMAIN>

		[Krb5LoginModule] authentication failed 
krb5.conf loading failed
YYYY-MM-DD HH:MM:SS,mmm WARN  [org.keycloak.events] (executor-thread-15) type=LOGIN_ERROR, realmId=<rId>, clientId=account-console, userId=<uId>, ipAddress=<ip>, error=invalid_user_credentials, auth_method=openid-connect, redirect_uri=http://<domain>:8080/realms/<realm>/account/#/, code_id=<cId>, username=elvirus, authSessionParentId=<aSPId>, authSessionTabId=<aSTId>
```
krb5.conf is in /etc/ and has a symlink in the keycloak/conf/ directory.
Am I missing something?

**Comment by hillout** [comment]
Hi, did you manage to solve the problem?

**Reply by El-Virus** [reply]
Unfortunately, I wasn't able to.

**Reply by ThoSap** [reply]
See my comment https://github.com/keycloak/keycloak/discussions/19734#discussioncomment-9534363

**Comment by ThoSap** [comment]
Disable "Use Kerberos for password authentication" in the realm **User federation** settings.
![image](https://github.com/keycloak/keycloak/assets/6555967/bcc5e691-0730-41cf-aa69-a11cf0dc24e7)

**Comment by oculos** [comment]
I had a few problems with keytabs before getting it to work.
First: do you know which encryptions you got into it? I saw that Keycloak is a bit picky about it.
Second: Could you have your keytab somewhere and point to it, instead of using symlinks?
